### ISSUE TITLE
Tapering fails when when the test particle is lost

### ISSUE THREAD

**Issue by anabramo** [xsuite/xsuite#430]
Recently, there was a check introduced in xtrack to verify compensation of energy loss (tapering) is required:

In xtrack/tapering.py:

    # Check whether compensation is needed
    p_test = particle_on_co.copy()
    p_test.delta = delta0
    line.track(p_test, turn_by_turn_monitor='ONE_TURN_EBE')
    mon = line.record_last_track
    eloss = -(mon.ptau[0, -1] - mon.ptau[0, 0]) * p_test.p0c[0]
    if abs(eloss) < p_test.energy0[0] * rtol_eneloss:
        if verbose: _print("  - No compensation needed")
        return

when the test particle is lost due to too much energy loss, like in the FCC-ee ttbar mode,  and the defaule delta0=0, the check with the monitor delta reports that no tapering is needed. Delta seems to be zero when the particle is dead, which is the source of the problem. Commenting out the check makes the tapering work as before.

**Comment by giadarol** [comment]
Thanks for reporting this!
We should probably change the check into:
```python
if p_test.state <=0 or abs(eloss) < p_test.energy0[0] * rtol_eneloss:
    if verbose: _print("  - No compensation needed")
    return
```

**Comment by giadarol** [comment]
@anabramo could you please check that the above works for your case?

**Comment by anabramo** [comment]
Thanks for the prompt reply. The suggested change works if I invert the state check:

    if p_test.state[0] > 0 and abs(eloss) < p_test.energy0[0] * rtol_eneloss:
        if verbose: _print("  - No compensation needed")
        return

e.g only skip the tapering if it alive and has no energy change

**Comment by giadarol** [comment]
yes indeed!

**Comment by giadarol** [comment]
Done
